### PR TITLE
check empty data dir

### PR DESCRIPTION
**Pull request by siddontang**
Data dir must be set explicitly, can't be empty. 

PTAL @BusyJay @zhangjinpeng1987

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
LGTM

File: src/storage/config.rs [hunk 1]
```diff
@@ -54,9 +54,11 @@ impl Default for Config {
 
 impl Config {
     pub fn validate(&mut self) -> Result<(), Box<Error>> {
-        if self.data_dir != DEFAULT_DATA_DIR {
-            self.data_dir = try!(config::canonicalize_path(&self.data_dir))
+        if self.data_dir.is_empty() {
```
Comment (BusyJay) [inline src/storage/config.rs]: Should comment why `DEFAULT_DATA_DIR` is not allowed when validating configuration.

Comment (siddontang) [inline src/storage/config.rs]: maybe it is better to remove DEFAULT_DATA_DIR and use "" directly in the default function?

Comment (BusyJay) [inline src/storage/config.rs]: `DEFAULT_DATA_DIR` is used for many storage tests.

Comment (siddontang) [inline src/storage/config.rs]: no, for storage tests, we all use temp directory.

**Comment by siddontang** [comment]
PTAL @BusyJay

File: src/storage/config.rs (hunk 1 above)
Comment (BusyJay) [inline src/storage/config.rs]: It checks whether if `config.data_dir` is equals to `TEMP_DIR`, which is empty, to decide using a temporary directory or specified path.

Comment (siddontang) [inline src/storage/config.rs]: where is the check? I just remove the default data dir and no compile error occurs.

Comment (BusyJay) [inline src/storage/config.rs]: https://github.com/pingcap/tikv/blob/master/src/storage/engine/rocksdb.rs#L87

Comment (siddontang) [inline src/storage/config.rs]: EngineRocksdb doesn't use this Config at all.

Comment (BusyJay) [inline src/storage/config.rs]: Yes. But storage use config to create an EngineRocksdb. See https://github.com/pingcap/tikv/blob/ef23163293faedf46e9b65f1991e572d72230a0c/src/storage/mod.rs#L436.

Comment (siddontang) [inline src/storage/config.rs]: Got it

**Pull request closed without merging**